src/ManagedServices/ManagedServices/Commands/GetAzureRmManagedServicesAssignment.cs

```diff
@@ -0,0 +1,107 @@
+    using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet(
+        VerbsCommon.Get,
+        Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesAssignment",
+        DefaultParameterSetName = DefaultParameterSet), OutputType(typeof(PSRegistrationAssignment))]
+    public class GetAzureRmManagedServicesAssignment : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+        protected const string ByResourceIdParameterSet = "ByResourceId";
+        protected const string ByIdParameterSet = "ById";
+
+        [Parameter(Mandatory = false, ParameterSetName = DefaultParameterSet, HelpMessage = "The scope where the registration assignment is created.")]
+        [Parameter(Mandatory = false, ParameterSetName = ByIdParameterSet, HelpMessage = "The scope where the registration assignment is created.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ByIdParameterSet, HelpMessage = "The Registration Assignment identifier.")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ByResourceIdParameterSet,
+            HelpMessage = "The fully qualified resource id of registration assignment.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, HelpMessage = "Whether to include registration definition details.")]
+        [Parameter(ParameterSetName = ByResourceIdParameterSet, HelpMessage = "Whether to include registration definition details.")]
+        [Parameter(ParameterSetName = ByIdParameterSet, HelpMessage = "Whether to include registration definition details.")]
+        public SwitchParameter ExpandRegistrationDefinition { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string scope = null;
+            string assignmentId = null;
+
+            if (this.IsParameterBound(x => x.ResourceId))
+            {
+                assignmentId = this.ResourceId.GetResourceName();
+                scope = this.ResourceId.GetSubscriptionId().ToSubscriptionResourceId();
+            }
+            else if (this.IsParameterBound(x => x.Id))
+            {
+                assignmentId = this.Id;
+                scope = this.Scope ?? this.GetDefaultScope();
+            }
+            else if (this.IsParameterBound(x => x.Scope))
+            {
+                scope = this.Scope;
+            }
+            else
+            {
+                scope = this.GetDefaultScope();
+            }
+
+            ConfirmAction(MyInvocation.InvocationName,
```

Comment:
You should not use ConfirmAction here - this is only appropriate when implementing ShouldProcess, and Gets should not support ShouldProcess